### DISCUSSION TITLE
Multiple UTXOs and multiple contract commitments in a transaction?

### DISCUSSION THREAD

**Discussion by eauxxs** [RGB-WG/rgb#206] [Q&A]
I know that multiple contracts can be associated with a utxo, and commitments can be made using tapret or opret in a transaction. Suppose there are two utxos, each associated with a contract, and I want to exchange their associated tokens in one transaction. If opret is used for commitment, two opreturn outputs will be generated in output, but this will not be allowed on the chain, which is an incorrect transaction.

And, because the names of opret1st and tapret1st intuitively show that the index of the commitment is a txid, and the specific location is the first taproot output or opreturn output, does this mean that only one utxo in a transaction can make rgb commitment?

**Comment by dr-orlovsky** [comment]
> If opret is used for commitment, two opreturn outputs will be generated in output

No it wan't. I have no idea where you got that.

> does this mean that only one utxo in a transaction can make rgb commitment?

No. Please read on how RGB is working - what you describe has nothing to do with the actual protocol.
